libbeat/metric/system/process/process_linux.go

```diff
@@ -0,0 +1,28 @@
+package process
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/elastic/gosigar"
+)
+
+// GetSelfPid return self pid for current running process
+func GetSelfPid() (int, error) {
+	contents, err := ioutil.ReadFile(gosigar.Procd + "/self/stat")
```

Comment:
I was just thinking that it could be enough with `os.Readlink(gosigar.Procd + "/self")`, then no more parsing would be needed. Could you try? Thanks!